core/trino-main/src/main/java/io/trino/execution/PrivilegeUtilities.java

```diff
@@ -44,6 +48,24 @@ public static Set<Privilege> parseStatementPrivileges(Node statement, Optional<L
         return privileges;
     }
 
+    public static Set<EntityPrivilege> parseEntityKindPrivileges(String entityKind, Metadata metadata, Optional<List<String>> privileges)
+    {
+        Set<EntityPrivilege> allPrivileges = metadata.getAllEntityKindPrivileges(entityKind);
+        if (privileges.isPresent()) {
+            return privileges.get().stream()
+                    .map(privilege -> {
+                        EntityPrivilege entityPrivilege = new EntityPrivilege(privilege.toUpperCase(Locale.ENGLISH));
+                        if (!allPrivileges.contains(entityPrivilege)) {
+                            throw new ParsingException("Privilege %s is not supported for entity kind %s".formatted(privilege, entityKind));
```

Comment:
This should not be a ParsingException, but a TrinoException with INVALID_PRIVILEGE error code.

Reply:
Changed.